### PR TITLE
#13691 errors to top workbench window

### PR DESCRIPTION
**Pull request by Destrolaric**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java [hunk 1]
```diff
@@ -1671,6 +1671,30 @@ public static IWorkbenchWindow getParentWorkbenchWindow(Control control) {
         return null;
     }
 
+    /**
+     * We try to find last visible child shell, because some Linux desktop environments are handling window modality differently
+     * see #13691
+     * @return last visible child shell
+     */
+    @Nullable
+    public static Shell getTopActiveWorkbenchShell(){
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java]:
```suggestion
    public static Shell getTopActiveWorkbenchShell() {
```

File: plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java [hunk 2]
```diff
@@ -1671,6 +1671,30 @@ public static IWorkbenchWindow getParentWorkbenchWindow(Control control) {
...
+        IWorkbench workbench = PlatformUI.getWorkbench();
+        Shell shell = workbench.getActiveWorkbenchWindow().getShell();
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java:1684]: Method `getActiveWorkbenchWindow` may return `null`

File: plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java [hunk 3]
```diff
@@ -1671,6 +1671,30 @@ public static IWorkbenchWindow getParentWorkbenchWindow(Control control) {
...
+        while (shell.getShells() != null){
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java]:
```suggestion
        while (shell.getShells() != null) {
```

File: plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java [hunk 4]
```diff
@@ -1671,6 +1671,30 @@ public static IWorkbenchWindow getParentWorkbenchWindow(Control control) {
...
+            Shell parent = shell;
+            for (Control child : shell.getShells()) {
+                if (child.isVisible()){
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java]:
```suggestion
                if (child.isVisible()) {
```

File: plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java [hunk 5]
```diff
@@ -1671,6 +1671,30 @@ public static IWorkbenchWindow getParentWorkbenchWindow(Control control) {
...
+                    shell = child.getShell();
+                    break;
+                }
+            }
+            if (parent == shell){
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java]:
```suggestion
            if (parent == shell) {
```

File: plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java [hunk 6]
```diff
@@ -1671,6 +1671,30 @@ public static IWorkbenchWindow getParentWorkbenchWindow(Control control) {
         return null;
     }
 
+    /**
+     * We try to find last visible child shell, because some Linux desktop environments are handling window modality differently
+     * see #13691
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java]: Don't describe the reason _why_ this method exists. Describe what it does instead.

**Review by ShadelessFox — CHANGES_REQUESTED** [review]

**Review by ShadelessFox — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java [hunk 7]
```diff
@@ -1671,6 +1671,34 @@ public static IWorkbenchWindow getParentWorkbenchWindow(Control control) {
...
     }
 
+    /**
+     * Starting from root shell iterates through visible child shells to find
+     * shell without any visible children
+     * see #13691
+     * @return child shell without visible children
+     */
+    @Nullable
+    public static Shell getTopActiveWorkbenchShell() {
```
Comment (flambchap) [inline plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java:1682]: I'm not sure if it's a good decision to put this method here. It currently has only one usage, and `UIUtils` is already quite bloated.

**Comment by serge-rider** [comment]
Strange behavior. It searhces for the first visible shell, not the topmost one.

**Comment by serge-rider** [comment]
Let's do it differently. We need current asctive shell, not the deepest one

**Pull request closed without merging**